src/main/java/com/google/api/codegen/metacode/InitCodeContext.java

```diff
@@ -31,6 +31,10 @@
   public enum InitCodeOutputType {
     SingleObject,
     FieldList,
+    // This is used in grpc testing for when a method accepts a field list but the grpc request
+    // object is needed. This helps to compare the the request object that was constructed by the
+    // method against the request object that is initialized.
+    FieldListAndRequestObject
```

Comment:
Theres a lot of reflection going on in the client when wrapping an api method which makes it really hard to use an out of the box mocking library. I made a custom class that will make it easier to mock the method.